sql/src/main/java/io/crate/execution/dml/upsert/FromRawInsertSource.java

```diff
@@ -23,14 +23,20 @@
 package io.crate.execution.dml.upsert;
 
 import org.elasticsearch.common.bytes.BytesArray;
-import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
 
 import java.io.IOException;
+import java.util.Map;
 
 public class FromRawInsertSource implements InsertSourceGen {
 
     @Override
-    public BytesReference generateSourceAndCheckConstraints(Object[] values) throws IOException {
-        return new BytesArray(((String) values[0]));
+    public Map<String, Object> generateSourceAndCheckConstraints(Object[] values) throws IOException {
```

Comment:
This change is unfortunate because it converts decodes the values again. Currently I am operating on the sources, so i need the decompressed values.